internal/inspectimage/info.go

```diff
@@ -0,0 +1,212 @@
+package inspectimage
+
+import (
+	"github.com/buildpacks/lifecycle"
+	"github.com/buildpacks/lifecycle/launch"
+
+	"github.com/buildpacks/pack"
+	"github.com/buildpacks/pack/internal/config"
+	"github.com/buildpacks/pack/internal/dist"
+)
+
+type GeneralInfo struct {
+	Name            string
+	RunImageMirrors []config.RunImage
+}
+
+type RunImageMirrorDisplay struct {
+	Name           string `json:"name" yaml:"name" toml:"name"`
+	UserConfigured bool   `json:"user_configured,omitempty" yaml:"user_configured,omitempty" toml:"user_configured,omitempty"`
+}
+
+type StackDisplay struct {
+	ID     string   `json:"id" yaml:"id" toml:"id"`
+	Mixins []string `json:"mixins,omitempty" yaml:"mixins,omitempty" toml:"mixins,omitempty"`
+}
+
+type ProcessDisplay struct {
+	Type    string   `json:"type" yaml:"type" toml:"type"`
+	Shell   string   `json:"shell" yaml:"shell" toml:"shell"`
+	Command string   `json:"command" yaml:"command" toml:"command"`
+	Default bool     `json:"default" yaml:"default" toml:"default"`
+	Args    []string `json:"args" yaml:"args" toml:"args"`
+}
+
+type BaseDisplay struct {
+	TopLayer  string `json:"top_layer" yaml:"top_layer" toml:"top_layer"`
+	Reference string `json:"reference" yaml:"reference" toml:"reference"`
+}
+
+type InfoDisplay struct {
+	StackID         string                  `json:"stack" yaml:"stack" toml:"stack"`
+	Base            BaseDisplay             `json:"base_image" yaml:"base_image" toml:"base_image"`
+	RunImageMirrors []RunImageMirrorDisplay `json:"run_images" yaml:"run_images" toml:"run_images"`
+	Buildpacks      []dist.BuildpackInfo    `json:"buildpacks" yaml:"buildpacks" toml:"buildpacks"`
+	Processes       []ProcessDisplay        `json:"processes" yaml:"processes" toml:"processes"`
+}
+
+type InspectOutput struct {
+	ImageName string       `json:"image_name" yaml:"image_name" toml:"image_name"`
+	Remote    *InfoDisplay `json:"remote_info" yaml:"remote_info" toml:"remote_info"`
+	Local     *InfoDisplay `json:"local_info" yaml:"local_info" toml:"local_info"`
+}
+
+func NewInfoDisplay(info *pack.ImageInfo, generalInfo GeneralInfo) *InfoDisplay {
+	if info == nil {
+		return nil
+	}
+	return &InfoDisplay{
+		StackID:         info.StackID,
+		Base:            displayBase(info.Base),
+		RunImageMirrors: displayMirrors(info, generalInfo),
+		Buildpacks:      displayBuildpacks(info.Buildpacks),
+		Processes:       displayProcesses(info.Processes),
+	}
+}
```

Comment:
nit: Seems like this is the main concept in this file. Can we rename it to `info_display.go`?